bindings/generate_bindings.py

```diff
@@ -1,341 +1,313 @@
+        pathstr = "f" + pathstr
+
+    if query_params:
+        out += ["    _params = {"]
+        for _, param in query_params:
+            if isinstance(param.type, swagger_types.Bool):
+                value = f"str({param.name}).lower()"
+                if not param.required:
+                    value += f" if {param.name} is not None else None"
+            elif need_parse(param.type):
+                value = f"{dump(param.type, param.name, omit_unset='True')}"
+                if not param.required:
+                    value += f" if {param.name} is not None else None"
+            else:
+                value = param.name
+            out += [f'        "{param.serialized_name}": {value},']
+        out += ["    }"]
+    else:
+        out += ["    _params = None"]
+
+    if "body" in anno.params:
+        # It is important that request bodies omit unset values so that PATCH request bodies
+        # do not include extraneous None values.
+        body_param = anno.params["body"]
+        bodystr = dump(body_param.type, body_param.name, "True")
+    else:
+        bodystr = "None"
+    out += ["    _resp = session._do_request("]
+    out += [f'        method="{anno.method.upper()}",']
+    out += [f"        path={pathstr},"]
+    out += ["        params=_params,"]
+    out += [f"        json={bodystr},"]
+    out += ["        data=None,"]
+    out += ["        headers=None,"]
+    out += ["        timeout=None,"]
+    out += [f"        stream={anno.streaming},"]
+    out += ["    )"]
+    for expect, returntype in anno.responses.items():
+        out += [f"    if _resp.status_code == {expect}:"]
+        is_none = isinstance(returntype, swagger_types.Ref) and (
+            returntype.defn is None
+            or isinstance(returntype.defn, swagger_types.Class)
+            and not returntype.defn.params
+        )
+        if not anno.streaming:
+            if is_none:
+                out += ["        return"]
+            else:
+                out += [f'        return {load(returntype, "_resp.json()")}']
+        else:
+            assert not is_none, "unable to stream empty result class: {anno}"
+            # Too many quotes to do bit inline:
+            yieldable = load(returntype, '_j["result"]')
+            out += [
+                f"        for _line in _resp.iter_lines():",
+                f"            _j = json.loads(_line)",
+                f'            if "error" in _j:',
+                f"                raise APIHttpStreamError(",
+                f'                    "{anno.method}_{anno.name}",',
+                f'                    runtimeStreamError.from_json(_j["error"])',
+                f"            )",
+                f"            yield {yieldable}",
+                f"        return",
+            ]
+    out += [f'    raise APIHttpError("{anno.method}_{anno.name}", _resp)']
+
+    return "\n".join(out)
+
+
+def gen_def(anno: swagger_types.TypeDef) -> Code:
```

Comment:
I'd recommend breaking this up into `gen_class()` and `gen_enum()`:

```python
def gen_class(cls: swagger_types.Class) -> Code:
    ...

def gen_enum(enm: swagger_types.Enum) -> Code:
    ...

def gen_def(typedef: swagger_types.TypeDef) -> Code:
    if isinstance(anno, swagger_types.Class):
        return gen_class(anno)
    if isinstance(anno, swagger_types.Enum):
        return gen_enum(anno)
    assert_never(anno)
```

Reason: gen_class() and gen_enum() are plenty complex beasts by themselves, without sandwiching them into the same function.